lib/interfaces/include/BMSDriverGroup.h

```diff
@@ -0,0 +1,332 @@
+     */
+    uint8_t get_address(int ic_index) { return address[ic_index]; }
+
+    /**
+     * @return IC_data container
+     */
+    IC_Buffers_s get_IC_buffer() { return *this->IC_buffer; }
+
+    /**
+     * @return 2 byte broadcast command
+     */
+    uint8_t *generate_broadcast_command(CMD_CODES_e cmd_code) {
+        uint8_t cmd[2] = {((uint8_t)cmd_code) >> 8, (uint8_t)cmd_code};
+        return cmd;
+    }
+
+    /**
+     * @return 2 byte address command
+     */
+    uint8_t *generate_address_command(CMD_CODES_e cmd_code, int address) {
+        uint8_t cmd[2] = {(uint8_t)(get_cmd_address(address) | (uint8_t)cmd_code >> 8), (uint8_t)cmd_code};
+        return cmd;
+    }
+
+    /**
+     * @return usable command address for LTC6811_2
+    */
+    uint8_t get_cmd_address(int address) { return 0x80 | (address << 3); }
+
+private:
+    /**
+     * Pointer to the PEC table we will use to calculate new PEC tables
+     */
+    uint16_t pec15Table[256];
+
+    /**
+     * We will need this for both models of the IC
+     * This determines where we get our signals from on the Arduino
+     * It can only be 9 or 10
+     * NOTE: needs to be initialized
+     */
+    int chip_select;
+
+    /**
+     * Stores number of ICs on the chip select line: 6
+     * NOTE: needs to be initialized
+     */
+    int ic_count;
+
+    /**
+     * Stores all 63 cells of data in a 2D array.
+     * There are 6 boards with 2 ICs each with a total of 12 ICs on ACU
+     * BUT since each instance of LTC6811 will only have half, then we will hold 6 sets of data points
+     * NOTE: THIS IS THE PROCESSED DATA
+     */
+    IC_Voltage_Temp_Data_s IC_data[total_ic / 2];
+
+    /**
+     * Stores Register Group Bytes, 6 bytes each
+     * There are many other register groups like PWM, S_Control, and COMM bytes,
+     * But we don't need that functionality for our purposes
+     * NOTE: THIS IS THE RAW DATA
+     */
+    IC_Buffers_s IC_buffer[total_ic / 2];
+
+    /**
+     * Stores where the max voltage location is
+     */
+    IC_Cell_Location_s max_voltage_location;
+
+    /**
+     * Stores where the max voltage location is
+     */
+    IC_Cell_Location_s min_voltage_location;
+
+    /**
+     * Stores locations of max min GPIO data
+     */
+    IC_Cell_Location_s max_board_temp_location;
+    IC_Cell_Location_s min_board_temp_location;
+    IC_Cell_Location_s max_thermistor_location;
+    IC_Cell_Location_s max_humidity_location;
+    IC_Cell_Location_s min_thermistor_location;
+
+    /**
+     * We will only end up using the address if this is a LTC6811-2
+     * NOTE: But if we are, we need to call a setup function to instatiate each with the correct addresses
+     * BMS Segments 1, 4, and 5 are on chip select 9
+     * BMS Segments 2, 3, and 6 are on chip select 10
+     * Those segments correspond to 2 ICs each, so the instance with chip_select 9
+     * Will have IC addresses: 0,1,6,7,8,9 | The rest are for chip_select 10
+     */
+    int *address;
```

Comment:
where is this memory? if this is supposed to be an array, please make it so. please look at the video about scope / memory lifetimes